public/app/plugins/datasource/dashboard/runSharedRequest.ts

```diff
@@ -49,11 +50,24 @@ export function runSharedRequest(options: QueryRunnerOptions): Observable<PanelD
     }
 
     const listenToRunner = listenToPanel.getQueryRunner();
-    const subscription = listenToRunner.getData({ withTransforms: false, withFieldConfig: false }).subscribe({
-      next: (data: PanelData) => {
-        subscriber.next(data);
-      },
-    });
+    const subscription = listenToRunner
+      .getData({
+        withTransforms: Boolean(query?.withTransforms),
+        withFieldConfig: false,
+      })
+      .subscribe({
+        next: (data: PanelData) => {
+          // Use annotation data for series
+          if (query?.topic === DataTopic.Annotations) {
+            data = {
+              ...data,
+              series: data.annotations ?? [],
+              annotations: undefined, // remove annotations
+            };
+          }
+          subscriber.next(data);
+        },
+      });
```

Comment:
Would be nice with new unit tests that cover this, but runSharedRequest did not have any to start from. But does not look that tricky.

Reply:
done 🎉

Reply:
nice!